noir-projects/aztec-nr/aztec/src/initializer.nr

```diff
@@ -23,3 +31,14 @@ pub fn compute_contract_initialization_nullifier<TContext>(context: TContext) ->
 pub fn compute_unsiloed_contract_initialization_nullifier<TContext>(context: TContext) -> Field where TContext: ContextInterface {
     context.this_address().to_field()
 }
+
+pub fn assert_initialization_args_match_address_preimage<TContext>(context: TContext) where TContext: ContextInterface {
+    let address = context.this_address(); 
+    let instance = get_contract_instance(address);
+    let expected_init = compute_initialization_hash(context.selector(), context.get_args_hash());
+    assert(instance.initialization_hash == expected_init, "Initialization hash does not match");
+}
+
+pub fn compute_initialization_hash(init_selector: FunctionSelector, init_args: Field) -> Field {
+    pedersen_hash([init_selector.to_field(), init_args], GENERATOR_INDEX__CONSTRUCTOR)
```

Comment:
Small nit. Looking at this function in isolation made me think there's only parameter passed for initilisation.

```suggestion
pub fn compute_initialization_hash(init_selector: FunctionSelector, init_args_hash: Field) -> Field {
    pedersen_hash([init_selector.to_field(), init_args_hash], GENERATOR_INDEX__CONSTRUCTOR)
```